json_test.go

```diff
@@ -59,12 +59,12 @@ func TestJSONUnmarshal(t *testing.T) {
 		},
 		"empty": {
 			data:           []byte(`{"ID1": ""}`),
-			expectedError:  invalidLengthError{len: 0},
+			expectedError:  ErrInvalidLength,
 			expectedResult: Nil,
 		},
 		"omitempty": {
 			data:           []byte(`{"ID2": ""}`),
-			expectedError:  invalidLengthError{len: 0},
+			expectedError:  ErrInvalidLength,
```

Comment:
This is concerning.  It implies that `json.Unmarshal` will now return a different error than it did before.  Previously it returned an error that contained the length of the source.  Are you sure this works?  The code is returning something the results of `fmt.Errorf` which is not `ErrInvalidLength`

Reply:
Solved in [54f9572](https://github.com/google/uuid/pull/166/commits/54f95728c20a700436b4f3a8e699a99235748175)